library/Director/Web/Form/Element/Dictionary.php

```diff
@@ -79,7 +74,7 @@ public function setValue($value)
         if ($value instanceof \stdClass) {
             $value = json_decode(json_encode($value), true);
         }
-        return parent::setValue($value);
+        return parent::setValue($this->applyOnStructure($value));
```

Comment:
Appart from having a validate that sets a value (!? topic for another another review)... you are already applying the structure to the value on line 19 and this is a protected method so is there a case where you will get validateNode invoked with a non structure-applied value?